### PR TITLE
Improve dataset conversion 

### LINKED ISSUE #72: List dataset used for training/testing the model

### PR DESCRIPTION
**Pull request by naga-karthik**
This PR improves the way how the dataset is converted to nnUNet format for training. Main changes: 

1. Instead of expecting input datasets to RPI-oriented, the re-orientation to RPI is done on-the-fly during dataset conversion (commit fc06bb4)
2. The dataset commits are now logged to both `logs.txt` and `dataset.json` and can be found in the `path_out` folder (commits 88f1dd3 and e9eba99)
3. The exact nifti files used for training/testing (for each seed) are now also saved to the `path_out` folder under the name `train_test_split_seed<>.yaml` (commit bff6672)

Examples screenshots of how the info look: 

<details>

<summary> dataset commit info </summary>

Screen Shot 2023-12-12 at 5 47 36 PM

</details>

<details>

<summary> train/test splits </summary>

```
train: 
- sub-045_T2w.nii.gz
- sub-046_T2w.nii.gz
- sub-5425_T2w.nii.gz
- sub-5445_T2w.nii.gz
- sub-zh07_ses-01_acq-sag_T2w.nii.gz
- sub-zh08_ses-01_acq-ax_T2w.nii.gz
- sub-zh08_ses-01_acq-sag_T2w.nii.gz
- sub-zh09_ses-01_acq-sag_T2w.nii.gz

test: 
- sub-6550_T2w.nii.gz
- sub-6564_T2w.nii.gz
- sub-6574_T2w.nii.gz
- sub-zh05_ses-01_acq-sag_T2w.nii.gz
- sub-zh11_ses-01_acq-sag_T2w.nii.gz
```

</details>

Resolves #72 and #61

### CODE REVIEW

**Review by valosekj — APPROVED** [review]
Haven't tried but LGTM! Thank you!
